### PR TITLE
Feat/15700919 verified members form

### PR DESCRIPTION
**Pull request by CzarekDryl**
https://tw.auroracreation.com/app/tasks/15700919?c=5894784

**New stuff** ✨

* Added verified members form (only UI)
* Added verified members select with multiple choices

Resolves [1570](https://github.com/JoinColony/colonyCDapp/issues/1570)

### CODE REVIEW

File: src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/consts.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        label: formatText({ id: 'actionSidebar.option.removeMembers' }),
+        value: AddRemoveMembersType.Remove,
+      },
+    ],
+  };
+};
+
+export const validationSchema = object()
+  .shape({
+    createdIn: string().defined(),
```
Comment (ArmandoGraterol) [inline src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/consts.ts:32]: Beware that if this PR: #1702 gets merged first then this should be updated to `number`. Otherwise, I'll update this field there if this one gets merged first.

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
Nice job with this!

- "Add members" option appears with the color of the placeholders:
Screenshot 2024-01-15 at 18 35 13

- The upper corners of the verified table seem to be cut-off:
Screenshot 2024-01-15 at 18 38 21

- The submit button should read "Update verified" instead of `Create Action`:
Screenshot 2024-01-15 at 18 40 47

File: src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/consts.ts (hunk 1 above)
Comment (ArmandoGraterol) [reply src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/consts.ts:32]: That other PR was merged so this should be updated to a `number` schema :)

Comment (CzarekDryl) [reply src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/consts.ts]: @ArmandoGraterol Changed to number :) Also this `The upper corners of the verified table seem to be cut-off:` is fixed in other PR

**Comment by CzarekDryl** [comment]
@melyndav Please take a look once again, I think that I've fixed all issues mentioned by you. Metadata will be added later when I'll wire up form with backend.

File: src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        : ActionTypes.ROOT_MOTION,
+    getFormOptions,
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    transform: useCallback(
+      pipe(
+        // @TODO: Fix types when form will be wired up with API
+        mapPayload((values: any) => {
+          if (!colony) {
+            return null;
+          }
```
Comment (ArmandoGraterol) [inline src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/hooks.ts:45]: With the recent changes to the ColonyContext types, this check shouldn't be necessary

File: src/i18n/en.json [hunk 3]
```diff
@@ -1296,6 +1311,9 @@
     "motion.finalizeStep.submit": "Finalize",
     "motion.finalizeStep.transactions.remaining": "{transactions} transactions remaining",
     "motion.finalizeStep.claimed": "Claimed",
+    "motion.outcomeStep.win.title": "Congratulations! Your side won.",
+    "motion.outcomeStep.lost.title": "Sorry, your side lost.",
```
Comment (ArmandoGraterol) [inline src/i18n/en.json:1315]: I think these might have slipped up on the commit as I see that they aren't being used anywhere.

**Review by ArmandoGraterol — COMMENTED** [review]
Looking good to me 👍 We just have to address some small things I've noticed but after that this should be good to go from my side

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @CzarekDryl,

Thanks for the updates.

A couple of more updates I've found:

1. When I add a member, the kebab option now longer shows for the 'Add members' option and not the 'Remove members'. Also not showing on mobile. 

image


2. The horizontal rule is still showing between table rows on desktop, this shouldn't show at all ever in any action type including this one. As you can see by the Figma design:

image
 
 
 3. I'm still seeing issues with mobile formatting especially with the action title 

image

4. With tables on mobile, can the rows widths be adjusted to more align with Figma? You can see the members columns are much narrower on mobile giving more space to the user title.

image


5. There is a problem with table corners on desktop and mobiles as Armando raised. 

6. When I add a new row on mobile, the kebab icons disappear on the 'remove members' variation

image

However when I add a new rows, the icon show again 

image

**Comment by bassgeta** [comment]
we can change the base branch to be https://github.com/JoinColony/colonyCDapp/pull/1773 since we're gonna merge everything into a feature branch first :D 
#1733 is the issue tracking this, also please check my implementation details just in case
otherwise good work on this

File: src/components/v5/common/ActionSidebar/partials/VerifiedMembersTable/VerifiedMembersTable.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import { formatText } from '~utils/intl';
+import TableWithMeatballMenu from '~v5/common/TableWithMeatballMenu';
+import Button from '~v5/shared/Button/Button';
+
+import { useVerifiedMembersTableColumns, useGetTableMenuProps } from './hooks';
+import { VerifiedMembersTableModel, VerifiedMembersTableProps } from './types';
+
+const displayName = 'v5.common.ActionsContent.partials.VerifiedMembersTable';
+
+const VerifiedMembersTable: FC<VerifiedMembersTableProps> = ({ name }) => {
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/VerifiedMembersTable/VerifiedMembersTable.tsx:17]: could we perhaps pass in 2 functions
1. `addMember` which would wrap around form logic and add to the array
2. `removeMember` which would wrap around form logic to remove members
3. the data also needs to get passed in and not get read from the form
4. can we name it just `MembersTable` since we use it to display members we will add as verified or remove from verified

File: src/components/v5/common/ActionSidebar/partials/VerifiedMembersTable/VerifiedMembersTable.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      )}
+      {!readonly && (
+        <Button
+          mode="primaryOutline"
+          iconName="plus"
+          size="small"
+          className="mt-6"
+          isFullSize={isMobile}
+          onClick={() => {
+            fieldArrayMethods.append({});
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/VerifiedMembersTable/VerifiedMembersTable.tsx:63]: could this just open the select by setting a boolean?

File: src/components/v5/common/ActionSidebar/partials/VerifiedMembersSelect/VerifiedMembersSelect.tsx [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+import UserAvatar from '~v5/shared/UserAvatar';
+
+import { AddRemoveMembersType } from '../forms/ManageVerifiedMembersForm/consts';
+
+import { useVerifiedMembersSelect } from './hooks';
+import { VerifiedMembersSelectProps } from './types';
+
+const displayName = 'v5.common.ActionsContent.partials.VerifiedMembersSelect';
+
+const VerifiedMembersSelect: FC<VerifiedMembersSelectProps> = ({
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/VerifiedMembersSelect/VerifiedMembersSelect.tsx:26]: can we split this one into 2 different ones, one for selecting from already verified members, the other one for selecting from unverified members?
we can reuse styles and everything, just 2 different components which will fetch from different sources
we can mock them until we get the BE up and running :)

File: src/components/v5/common/ActionSidebar/partials/VerifiedMembersSelect/VerifiedMembersSelect.tsx [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+  const { readonly } = useAdditionalFormOptionsContext();
+  const isMobile = useMobile();
+
+  const { portalElementRef, relativeElementRef } = useRelativePortalElement<
+    HTMLButtonElement,
+    HTMLDivElement
+  >([isUserSelectVisible]);
+  const showOnlyNotVerifiedMembers =
+    addRemoveMembers === AddRemoveMembersType.Add;
+  const { getValues, setValue } = useFormContext();
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/VerifiedMembersSelect/VerifiedMembersSelect.tsx:59]: I would decouple this from all form related things, pass in the data from the form, it's cleaner that way

File: src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/consts.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
+import { array, InferType, number, object, string } from 'yup';
+
+import { MAX_ANNOTATION_LENGTH } from '~constants';
+import { formatText } from '~utils/intl';
+import { ACTION_BASE_VALIDATION_SCHEMA } from '~v5/common/ActionSidebar/consts';
+import { CardSelectOption } from '~v5/common/Fields/CardSelect/types';
+
+export enum AddRemoveMembersType {
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/consts.ts:8]: maybe we can rephrase all `AddRemove` into `Manage`

File: src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/hooks.ts [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+  useActionFormBaseHook({
+    actionType:
+      decisionMethod === DecisionMethod.Permissions
+        ? ActionTypes.ACTION_VERSION_UPGRADE
+        : ActionTypes.ROOT_MOTION,
+    getFormOptions,
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    transform: useCallback(
+      pipe(
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/hooks.ts:40]: do we need a `pipe` here?

File: src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/hooks.ts [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        // @TODO: Fix types when form will be wired up with API
+        mapPayload((values: any) => {
+          if (!colony) {
+            return null;
+          }
+
+          return getManageWhitelistDialogPayload(
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/hooks.ts:47]: we can stub this out or jsut write a comment because we will not be using this function

File: src/components/v5/common/Fields/CardSelect/FormCardSelect.tsx [hunk 11]
```diff
@@ -12,11 +12,12 @@ const displayName = 'v5.common.Fields.CardSelect.FormCardSelect';
 
 function FormCardSelect<TValue = string>({
   name,
+  onChange,
```
Comment (bassgeta) [inline src/components/v5/common/Fields/CardSelect/FormCardSelect.tsx:15]: why are these changes necessary? if it's a form component, we can do this by reading from form values?

File: src/components/v5/shared/SearchSelect/partials/SearchItem/SearchItem.tsx [hunk 12]
```diff
@@ -39,6 +64,7 @@ const SearchItem: FC<SearchItemProps> = ({
           showAvatar,
           color,
           missingPermissions,
+          isVerified,
```
Comment (bassgeta) [inline src/components/v5/shared/SearchSelect/partials/SearchItem/SearchItem.tsx:67]: we are adding `verifiedMembers` specific things into a reusable component, can we compose that checkbox in when defining the items?

File: src/images/icons.json [hunk 13]
```diff
@@ -192,6 +192,7 @@
     "phrase",
     "plus",
     "plus-heavy",
+    "plus-minus",
```
Comment (bassgeta) [inline src/images/icons.json:195]: can we not do this and instead use the icon from `phosphor`?

**Review by bassgeta — CHANGES_REQUESTED** [review]

File: src/images/icons.json (hunk 13 above)
Comment (CzarekDryl) [reply src/images/icons.json:195]: It's being passed to `ActionFormRow` and it's used in all forms, so I think it will be good to refactor all icons used there but maybe in other PR since this one is getting messy?

Comment (bassgeta) [reply src/images/icons.json:195]: I think that component has support for passing in a rendered component now so we can try and see if it works
image

File: src/components/v5/common/Fields/CardSelect/FormCardSelect.tsx (hunk 11 above)
Comment (CzarekDryl) [reply src/components/v5/common/Fields/CardSelect/FormCardSelect.tsx:15]: It was added to pass a reset function, but with changes that you requested I don't think it will be needed

Comment (CzarekDryl) [reply src/components/v5/common/Fields/CardSelect/FormCardSelect.tsx:15]: It's additional onChange, I've passed a form reset function there.

File: src/images/icons.json (hunk 13 above)
Comment (CzarekDryl) [reply src/images/icons.json:195]: It worked, so I've changed all icons in this table to phosphor

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @CzarekDryl, thanks again for the further updates.

A few more _UI only_ issues on this one.

1. The kebab icons are not center aligned in the table rows.
image

2. Table row heights do not match Figma heights
image

3. The kebab menu should be style like the other mobile components for this feature:

image
 
4. Can we set a max height on this member pop over as per Figma? Currently it changes height and just fills the rest of the vertical screen height on desktop. It would be great to have this consistent across all action types.

File: src/components/v5/common/Fields/CardSelect/FormCardSelect.tsx (hunk 11 above)
Comment (bassgeta) [reply src/components/v5/common/Fields/CardSelect/FormCardSelect.tsx:15]: can this be done outside of this component? maybe when a value changes, we can reset the form, not overriding this onChange

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Nice job so far on this! I’ve picked up a few issues whilst testing - not sure if they’re all within the scope of this PR.

If you remove the last member from the “selected members” form, everything will crash. This happens whether you remove via the kebab menu, or via the select dropdown.

https://github.com/JoinColony/colonyCDapp/assets/34915414/f6fdcf9d-b401-4172-b8bf-00cb219c5877

https://github.com/JoinColony/colonyCDapp/assets/34915414/ef2f6f3e-b48b-4ece-a8f6-20cdd8aa8f8a

The blockies here don’t match between users in the form vs the dropdown menu. #1712 recently resolved this elsewhere in the app so would be great to keep that consistent.

Screenshot 2024-01-29 at 11 33 10

It is impossible to scroll to the bottom of the select dropdown menu if it is close to the bottom of the screen (desktop and mobile).

https://github.com/JoinColony/colonyCDapp/assets/34915414/4b5fdac6-43ae-43d3-ad64-7af0998cfaf2

When adding a member it will briefly show the incorrect avatar and username.

https://github.com/JoinColony/colonyCDapp/assets/34915414/cefe6401-c804-488a-971b-442eb827c6bb

I’ve noticed similar issues on other action types, but the error naming here should be improved.

Screenshot 2024-01-29 at 11 29 46

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @CzarekDryl only a very minor UI request is the close in the mobile modal is the wrong colour. If you could update that, it would be much appreciated since this will be prevalent across most of the mobiles, thank you! 

image

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Thanks for making those changes, it’s definitely looking better! Blockies are matching up now, no longer seeing the incorrect avatar / username show briefly and error messaging is better. So this is great!

The page no longer crashes when removing the last member, but the table does disappear. It would be better if this could just stay as it is when the action is first selected.

https://github.com/JoinColony/colonyCDapp/assets/34915414/e36cd52b-3f14-4d53-9de5-fe32437a3295

The add / cancel buttons on mobile are also a bit misleading. Cancel implies that if clicked it will undo any changes I’ve made in the selection. And I’m able to deselect users and click “add” which is confusing too. I’m not sure what the best solution here is so I’ll let @melyndav weigh in.

https://github.com/JoinColony/colonyCDapp/assets/34915414/1cd6aabd-0299-4164-a96a-5ca8f5a817b6

**Comment by melyndav** [comment]
> The add / cancel buttons on mobile are also a bit misleading. Cancel implies that if clicked it will undo any changes I’ve made in the selection. And I’m able to deselect users and click “add” which is confusing too. I’m not sure what the best solution here is so I’ll let @melyndav weigh in.
> Screen.Recording.2024-02-02.at.10.17.58.mov

Hey @CzarekDryl the mobile menu has been designed so it doesn't have a cancel button for this action type only since the check boxes are present. 

image

Link to Figma design system component: https://www.figma.com/file/l1dOM5qiQYwF0ElvKDqqjg/Design-System---Colony-v3?type=design&node-id=1818-725&mode=design&t=blzmU3Z0fqsjGXuj-4

File: src/i18n/en.json (hunk 3 above)
Comment (ArmandoGraterol) [reply src/i18n/en.json:1348]: These messages still need to be removed as they are not being used anywhere.

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Thanks for making some more progress on this issue, it is getting there! The mobile menu is working a lot better now.

I've noticed a few new issues.

The members select list is missing the verified icons. I think it would look neater on mobile if the width matched the table.

Screenshot 2024-02-26 at 09 57 36

I think the button to open this menu just shouldn't show when there is only 1 row.

Screenshot 2024-02-26 at 09 58 32

The "Create action" button should read "Update verified".

Screenshot 2024-02-26 at 09 53 47

The "Created in" field should only show when the decision method is reputation.

Screenshot 2024-02-26 at 09 59 10

Lastly, the selected members list should reset when switching between add / remove, otherwise you get behaviour like this:

https://github.com/JoinColony/colonyCDapp/assets/34915414/b828676b-6177-4de2-a050-4c2c5e447a6d

File: src/components/common/ColonyActions/helpers/getActionTitleValues.ts [hunk 14]
```diff
@@ -104,6 +106,12 @@ const getMessageDescriptorKeys = (actionType: AnyActionType) => {
         ActionTitleMessageKeys.Recipient,
         ActionTitleMessageKeys.Initiator,
       ];
+    case actionType.includes(ExtendedColonyActionType.UpdateAddressBook):
```
Comment (bassgeta) [inline src/components/common/ColonyActions/helpers/getActionTitleValues.ts:109]: this is not necessary, maybe we can reuse the [strings](https://github.com/JoinColony/colonyCDapp/blob/bcf30454ca410212642c4c0b54e0fcb8880987f4/src/components/common/ColonyActions/helpers/getActionTitleValues.ts#L125) implemented for displaying titles in the action table?

File: src/components/v5/common/ActionSidebar/partials/MembersTable/hooks.tsx [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+            />
+          ) : (
+            <VerifiedMembersSelect
+              fieldArrayMethods={fieldArrayMethods}
+              name={`${name}.${row.index}`}
+            />
+          ),
+      }),
+    ],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/MembersTable/hooks.tsx:45]: can you drop a note on why we're disabling this rule?

File: src/components/v5/common/ActionSidebar/partials/NonVerifiedMembersSelect/NonVerifiedMembersSelect.tsx [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+  const { usersOptions } = useVerifiedMembersSelect();
+  const [
+    isUserSelectVisible,
+    {
+      toggle: toggleUserSelect,
+      toggleOff: toggleOffUserSelect,
+      registerContainerRef,
+    },
+  ] = useToggle();
+  const { user: userByAddress, loading: userByAddressLoading } =
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/NonVerifiedMembersSelect/NonVerifiedMembersSelect.tsx:42]: is there an option to store the entire member object when selecting? now we're fetching the members, storing their address in form values and then refetching them again

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Thank you for the changes! I've only got one other change. If you add new rows with the "Add member" button, it is now impossible to remove the empty rows due to the button change. Can you make it so that it reappears when there is more than 1 empty row.

Screenshot 2024-02-27 at 10 14 04

File: src/components/v5/common/ActionSidebar/partials/NonVerifiedMembersSelect/hooks.ts [hunk 17]
```diff
@@ -0,0 +1,53 @@
+import { useMemo } from 'react';
+
+import { useColonyContext } from '~context/ColonyContext.tsx';
+import { useGetNotVerifiedMembersQuery } from '~gql';
+import { type SearchSelectOption } from '~v5/shared/SearchSelect/types.ts';
+
+export const useVerifiedMembersSelect = () => {
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/NonVerifiedMembersSelect/hooks.ts:7]: i think this hook should be called `useNonVerifiedMembersSelect`

File: src/components/v5/common/ActionSidebar/partials/VerifiedMembersSelect/VerifiedMembersSelect.tsx [hunk 18]
```diff
@@ -0,0 +1,174 @@
...
+  const { usersOptions } = useVerifiedMembersSelect();
+  const [
+    isUserSelectVisible,
+    {
+      toggle: toggleUserSelect,
+      toggleOff: toggleOffUserSelect,
+      registerContainerRef,
+    },
+  ] = useToggle();
+  const { user: userByAddress, loading: userByAddressLoading } =
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/VerifiedMembersSelect/VerifiedMembersSelect.tsx:42]: is there an option to store the entire member object when selecting? now we're fetching the members, storing their address in form values and then refetching them again

File: src/components/v5/common/Table/Table.tsx [hunk 19]
```diff
@@ -145,6 +149,13 @@ const Table = <T,>({
                           border-r
                           border-gray-200
                         `}
+                        style={{
```
Comment (bassgeta) [inline src/components/v5/common/Table/Table.tsx:152]: do we need this? why are we comparing this to 150px specifically?

File: src/utils/colonyActions.ts [hunk 20]
```diff
@@ -98,8 +98,10 @@ export const getDetailItemsKeys = (actionType: AnyActionType) => {
         ActionPageDetails.Description,
       ];
     }
+    case actionType.includes(ExtendedColonyActionType.UpdateAddressBook): {
```
Comment (bassgeta) [inline src/utils/colonyActions.ts:101]: i think we can stop using this outdated action type

File: src/components/v5/shared/SearchSelect/partials/SearchItem/SearchItem.tsx [hunk 21]
```diff
@@ -71,13 +96,26 @@ const SearchItem: FC<SearchItemProps> = ({
...
                   },
                 )}
+                name={value.toString()}
                 onClick={() => {
                   onChange?.(value);
                 }}
+                withoutButton={!!checkboxesList}
               >
                 <div className="relative w-full flex items-center">
+                  {checkboxesList && (
```
Comment (bassgeta) [inline src/components/v5/shared/SearchSelect/partials/SearchItem/SearchItem.tsx:109]: hmmm looking at all the conditional rendering in this file makes me think if it would be beneficial to create a `CheckboxSearchItem` or something along those lines, so we can render that from `SearchSelect` and leave this component intact.

**Review by bassgeta — CHANGES_REQUESTED** [review]
Great progress on this, dropped a few comments that would be nice to resolve before we merge it.
1. Why is there no meatball menu to remove the users from the table?
image

2. If no members are selected, we display a meatball menu for removing the row, and nothing really happens when we click it
image

3. The meatball menu breaks into new lines on hover
image

File: src/components/v5/common/Table/Table.tsx (hunk 19 above)
Comment (CzarekDryl) [reply src/components/v5/common/Table/Table.tsx:157]: We need this to set column width, value of 150px is here because it's default value of this library

File: src/components/common/ColonyActions/helpers/getActionTitleValues.ts [hunk 22]
```diff
@@ -32,6 +32,8 @@ export enum ActionTitleMessageKeys {
   ToDomain = 'toDomain',
   TokenSymbol = 'tokenSymbol',
   ChainName = 'chainName',
+  Operation = 'operation',
```
Comment (bassgeta) [inline src/components/common/ColonyActions/helpers/getActionTitleValues.ts:35]: I think both of these are unused and can be removed

File: amplify/backend/api/colonycdapp/schema.graphql [hunk 23]
```diff
@@ -745,12 +745,10 @@ enum ColonyActionType {
   An action related to adding verified members
   """
   ADD_VERIFIED_MEMBERS
-  ADD_VERIFIED_MEMBERS_MOTION
```
Comment (bassgeta) [inline amplify/backend/api/colonycdapp/schema.graphql:748]: this should stay in

File: amplify/backend/api/colonycdapp/schema.graphql [hunk 24]
```diff
@@ -745,12 +745,10 @@ enum ColonyActionType {
...
   """
   An action related to removing verified members
   """
   REMOVE_VERIFIED_MEMBERS
-  REMOVE_VERIFIED_MEMBERS_MOTION
```
Comment (bassgeta) [inline amplify/backend/api/colonycdapp/schema.graphql:753]: this should stay in

File: src/components/v5/common/ActionSidebar/partials/NonVerifiedMembersSelect/NonVerifiedMembersSelect.tsx [hunk 25]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    fieldArrayMethods.append({ value });
+  };
+
+  return (
+    <div className="sm:relative w-full flex items-center">
+      {readonly ? (
+        <UserAvatar
+          user={selectedMember || field.value.value}
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/NonVerifiedMembersSelect/NonVerifiedMembersSelect.tsx:91]: could we save `field.value.value` to a variable called `walletAddress` and use that everywhere?

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
The UI looks great now, all approved on that side! But it looks like something is up with the rebasing as there are a bunch of edited files that shouldn't have been touched by this PR including the graphql schema, sagas, etc.

File: src/components/v5/common/ActionSidebar/partials/VerifiedMembersSelect/VerifiedMembersSelect.tsx [hunk 26]
```diff
@@ -0,0 +1,173 @@
...
+
+    fieldArrayMethods.append({ value });
+  };
+
+  return (
+    <div className="sm:relative w-full flex items-center">
+      {readonly ? (
+        <>
+          <UserAvatar
+            user={selectedMember || field.value.value}
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/VerifiedMembersSelect/VerifiedMembersSelect.tsx:92]: could we save `field.value.value` to a variable called `walletAddress` and use that everywhere?

File: src/components/v5/frame/ColonyHome/TmpRemoveVerifiedMembers.tsx [hunk 27]
```diff
@@ -33,25 +33,8 @@ export const TmpRemoveVerifiedMembers = () => {
       }),
     ),
   });
-  const unverifyMembersMotion = useAsyncFunction({
```
Comment (bassgeta) [inline src/components/v5/frame/ColonyHome/TmpRemoveVerifiedMembers.tsx:36]: if you are modifying these 2 components, you can just remove both of them, since the form is the end game, these `Tmp` components are just for testing purposes

File: src/components/v5/shared/SearchSelect/SearchSelect.tsx [hunk 28]
```diff
@@ -124,77 +131,80 @@ const SearchSelect = React.forwardRef<HTMLDivElement, SearchSelectProps>(
...
-                                <CaretUp size={12} />
+              {filteredList.length > 0
+                ? filteredList.map(({ options, title, isAccordion, key }) =>
+                    isAccordion ? (
+                      <div className="mb-[0.625rem] last:mb-0" key={key}>
+                        <div className="flex items-center justify-between px-2">
+                          <h5 className="text-4 text-gray-400 mb-2 uppercase">
+                            {formatText(title)}
+                          </h5>
+                          {isAccordion && (
```
Comment (bassgeta) [inline src/components/v5/shared/SearchSelect/SearchSelect.tsx:142]: we don't need this `isAccordion` check since it's done above

File: src/components/v5/shared/SearchSelect/hooks.ts [hunk 29]
```diff
@@ -7,12 +8,12 @@ import { type SearchSelectOptionProps } from './types.ts';
 export const useSearchSelect = (
   items: SearchSelectOptionProps[],
   searchValue: string,
+  showSearchValueAsOption: boolean,
```
Comment (bassgeta) [inline src/components/v5/shared/SearchSelect/hooks.ts:11]: do we need this property? is there any place where we wouldn't want to display the searched property? worst case, we can just filter the items from that component

File: src/redux/sagas/actions/addVerifiedMembers.ts [hunk 30]
```diff
@@ -1,46 +1,43 @@
...
+import type ColonyManager from '~context/ColonyManager';
+import { type Action, type AllActions, ActionTypes } from '~redux';
 import { transactionAddParams } from '~redux/actionCreators/transactions.ts';
 
 import {
   createTransaction,
   createTransactionChannels,
-  waitForTxResult,
-} from '../transactions/index.ts';
+  getTxChannel,
```
Comment (bassgeta) [inline src/redux/sagas/actions/addVerifiedMembers.ts:11]: this entire file seems like a bad rebase, please revert it to the way it's on the feature branch

File: src/redux/sagas/actions/removeVerifiedMembers.ts [hunk 31]
```diff
@@ -32,13 +31,7 @@ function* removeVerifiedMembersAction({
...
   meta,
 }: Action<ActionTypes.ACTION_REMOVE_VERIFIED_MEMBERS>) {
-  const batchKey = 'removeVerifiedMembers';
-
-  const { removeVerifiedMembers, annotateRemoveVerifiedMembers } =
-    yield createTransactionChannels(metaId, [
-      'removeVerifiedMembers',
-      'annotateRemoveVerifiedMembers',
-    ]);
+  let txChannel;
```
Comment (bassgeta) [inline src/redux/sagas/actions/removeVerifiedMembers.ts:34]: this entire file seems like a bad rebase, please revert it to the way it's on the feature branch

**Review by bassgeta — CHANGES_REQUESTED** [review]
There may be a mis-rebase here, since the add and remove verified members sagas seem reverted to a state of some commits ago

File: src/redux/types/actions/colonyActions.ts [hunk 32]
```diff
@@ -332,42 +334,6 @@ export type ColonyActionsActionTypes =
...
-        customActionTitle: string;
-      },
-      MetaWithSetter<object>
-    >
-  | ErrorActionType<ActionTypes.MOTION_REMOVE_VERIFIED_MEMBERS_ERROR, object>
-  | UniqueActionType<
-      ActionTypes.MOTION_REMOVE_VERIFIED_MEMBERS_SUCCESS,
-      object,
-      object
-    >
```
Comment (iamsamgibbs) [inline src/redux/types/actions/colonyActions.ts:370]: This shouldn't be removed

File: src/redux/sagas/utils/verifiedMembers.ts [hunk 33]
```diff
@@ -0,0 +1,25 @@
+enum ActionType {
```
Comment (bassgeta) [inline src/redux/sagas/utils/verifiedMembers.ts:1]: this file should not exist anymore

**Review by bassgeta — CHANGES_REQUESTED** [review]
just a file leftover, otherwise great job!

File: src/redux/sagas/utils/verifiedMembers.ts
Comment (iamsamgibbs) [inline src/redux/sagas/utils/verifiedMembers.ts:1]: I don't believe this file is necessary anymore as it has been moved to src/redux/sagas/utils/metadataDelta.ts

File: src/components/v5/shared/SearchSelect/partials/CheckboxSearchItem/index.ts
Comment (iamsamgibbs) [inline src/components/v5/shared/SearchSelect/partials/CheckboxSearchItem/index.ts:1]: Can remove this empty file

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Good job fixing up the rebase, I’ve left a two comments on files that can be removed then I think it is good to go. Thanks for moving the motion action types to their correct file too.

Edge case we should be aware of, if you add a new verified user, then immediately reopen the form to add / remove them, the same user shows under both selection lists. (Is fixed on a refresh, so I’m guessing the lists are cached)

https://github.com/JoinColony/colonyCDapp/assets/34915414/aa64406b-2c28-4089-b875-39371796b96a

Just flagging this here, but the verified members motions do not submit correctly - I suspect there is an issue with the payload, but that may be on our side to fix.

**Comment by CzarekDryl** [comment]
@iamsamgibbs @bassgeta DomainId added to payload

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Thanks for the fix, motions are working now! 🎉

One last thing, the completed action for removing via motion hasn't been hooked up (all good for permissions and adding members)

Screenshot 2024-03-04 at 09 32 14

**Comment by bassgeta** [comment]
> Thanks for the fix, motions are working now! 🎉
> 
> One last thing, the completed action for removing via motion hasn't been hooked up (all good for permissions and adding members)
> 
> Screenshot 2024-03-04 at 09 32 14

That was the feature branch's fault, it was missing a line in the `CompletedAction` switch statement. I've pushed the fix up right now and after a rebase it should work. Nice catch :muscle:

**Review by bassgeta — APPROVED** [review]
Tested all four cases out and they work, actions display correctly, and I wasn't able to break the form by adding/removing at random. Good job :muscle:

**Review by iamsamgibbs — APPROVED** [review]
All looks good now! Thank you for your work on this issue!
